hotel_elasticsearch/backup.py

```diff
@@ -1,36 +1,193 @@
+        if response.json()['_shards']['total'] > 0:
+            return False
+        return True
+
+    def restore_repository_exists(self):
+        try:
+            result = requests.get('http://localhost:9200/_snapshot/cluster_restore')
+            result.raise_for_status()
+            return True
+        except HTTPError as e:
+            logger.exception(e)
+            return False
+
+    def configure_backup(self):
+        self._configure_backup_repository()
+        self._configure_backup_schedule()
+
+    def _configure_backup_repository(self):
+        try:
+            result = requests.put(
+                'http://localhost:9200/_snapshot/cluster_backup',
+                json={
+                        'type': 's3',
+                        'settings': {
+                            'bucket': self.bucket,
+                            'base_path': self.cluster_node.backup_path,
+                            'max_snapshot_bytes_per_sec': '200mb',
+                        }
+                }
+            )
+            result.raise_for_status()
+        except HTTPError as e:
+            logger.exception(e)
+            # Even though the request fails the repository is still created and should be deleted.
+            requests.delete('http://localhost:9200/_snapshot/cluster_backup')
+            raise BackupException('Could not configure backup repository')
+
+    def configure_restore_repository(self, backup_path, bucket=None):
+        if bucket is None:
+            bucket = self.bucket
+        try:
+            result = requests.put(
+                'http://localhost:9200/_snapshot/cluster_restore',
+                json={
+                        'type': 's3',
+                        'settings': {
+                            'bucket': bucket,
+                            'base_path': backup_path,
+                            'readonly': True
+                        }
+                }
+            )
+            result.raise_for_status()
+        except HTTPError as e:
+            logger.exception(e)
+            # Even though the request fails the repository is still created and should be deleted.
+            requests.delete('http://localhost:9200/_snapshot/cluster_restore')
+            raise BackupException('Could not configure restore repository')
+
+    def _configure_backup_schedule(self):
         try:
-            with Timer('es_backup', days=1) as t:
-                bm = BackupManager()
-                try:
-                    bm.run_backup()
-                except HTTPError as e:
-                    logger.exception(e)
-                    t.clear_timer()  # If backup failed, try again on this or another node
-        except BlockedByAnotherTimerException:  # Backup has already been run, let it pass
-            pass
-        time.sleep(random.randrange(60, 120))
+            result = requests.put(
+                'http://localhost:9200/_slm/policy/nightly-backup',
```

Comment:
We should define this as some constant to avoid typos 😅

Reply:
RIght, done